src/gridiron/controls/ui/ContentValue.cpp

```diff
@@ -4,10 +4,9 @@ namespace GridIron {
     namespace Controls {
         namespace UI {
             ContentValue::ContentValue(std::shared_ptr<Control> parent, const std::string &key, std::string value) :
-                    Control(std::string("value-").append(key).c_str(), parent),
-                    This{shared_from_this()},
-                    key{key, (*this)},
-                    value{value, (*this)}
+                Control(std::string("value-").append(key).c_str(), parent),
+                key{This(), key},
+                value{This(), value}
```

Comment:
It looks like this is still calling shared_from_this() in a constructor, but now indirectly via This().  You can't use shared_from_this() until all the constructors are finished.

Reply:
Hm. I see. Do you see any neat tricks to avoid this?

Reply:
I'm afraid I don't have any quick fixes in mind.  Maybe rethink how you're using shared_ptr generally... like whether you actually need owning pointers in the various places you're using them.  Think about the lifetime of the objects...  If you really need to, you could have another function to call after construction which does the assignments of the necessary shared_ptrs, but that could be messy and not sure it's required.

Reply:
Ok, I was trying to be overly "correct" and in so doing, went overly complex I guess. I'm not sure what the standard is in C++17 anymore- I thought old school pointers were passé

Reply:
'Old school' pointers are still perfectly fine for non-owning pointers.  It's when you need ownership that you need unique_ptr, or for shared ownership, shared_ptr.  I don't think that's changed since C++11.

Reply:
Thanks again